cmd/contour/serve.go

```diff
@@ -384,51 +376,53 @@ func (s *Server) doServe() error {
 	if contourConfiguration.LeaderElection.DisableLeaderElection {
 		contourHandler.IsLeader = disableLeaderElection(s.log)
 	} else {
-		contourHandler.IsLeader = setupLeadershipElection(&s.group, s.log, contourConfiguration.LeaderElection, s.clients, contourHandler.UpdateNow)
+		contourHandler.IsLeader = setupLeadershipElection(&s.group, s.log, contourConfiguration.LeaderElection, s.coreClient, contourHandler.UpdateNow)
 	}
 
 	// Start setting up StatusUpdateHandler since we need it in
 	// the Gateway API controllers. Will finish setting it up and
 	// start it later.
 	sh := k8s.StatusUpdateHandler{
-		Log:       s.log.WithField("context", "StatusUpdateHandler"),
-		Clients:   s.clients,
-		Cache:     s.mgr.GetCache(),
-		Converter: converter,
-	}
-
-	// Inform on DefaultResources.
-	for _, r := range k8s.DefaultResources() {
-		if err := informOnResource(s.clients, r, eventHandler, s.mgr.GetCache()); err != nil {
-			s.log.WithError(err).WithField("resource", r).Fatal("failed to create informer")
+		Log:    s.log.WithField("context", "StatusUpdateHandler"),
+		Client: s.mgr.GetClient(),
+	}
+
+	// Inform on default resources.
+	for name, r := range map[string]client.Object{
+		"httpproxies":               &contour_api_v1.HTTPProxy{},
+		"tlscertificatedelegations": &contour_api_v1.TLSCertificateDelegation{},
+		"extensionservices":         &contour_api_v1alpha1.ExtensionService{},
+		"contourconfigurations":     &contour_api_v1alpha1.ContourConfiguration{},
+		"services":                  &corev1.Service{},
+		"ingresses":                 &networking_v1.Ingress{},
+		"ingressclasses":            &networking_v1.IngressClass{},
+	} {
```

Comment:
it seemed easier to just inline these now that they're one-liners each, but if you all prefer this being kept in `informers.go` in a `DefaultResources()` function I can go back to that.

Reply:
Agree inline is cleaner.